src/main/scala/sbtassembly/Assembly.scala

```diff
@@ -219,24 +217,30 @@ object Assembly {
   // which jars exactly belong to the deps for packageDependency option.
   def assembleMappings(classpath: Classpath, dependencies: Classpath,
       ao: AssemblyOption, log: Logger): Vector[MappingSet] = {
-    val tempDir = ao.assemblyDirectory.get
-    if (!ao.cacheUnzip) IO.delete(tempDir)
```

Comment:
I should double check the original delete code, not sure if the `cacheOutput` was being used?